test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go

```diff
@@ -52,30 +76,436 @@ var _ = Describe("[performance] Cgroups and affinity", Ordered, func() {
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testpod1.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
+
+				By("Getting the container cpuset.cpus cgroup")
+				testpod2Cpus, err := pods.ExecCommandOnPod(testclient.K8sClient, testpod2, tasksetcmd)
+				testlog.Infof("%v pod is using %v cpus", testpod2.Name, string(testpod2Cpus))
+
+				// Get cpus used by the ovnkubenode-pods containers
+				// Each kubenode pods have many containers, we check cpus of only 1 container
+				ovnContainerCpus, err := getOvnContainerCpus(workerRTNode)
+				Expect(err).ToNot(HaveOccurred(), "Unable to fetch cpus of the containers inside ovn kubenode pods")
+				testlog.Infof("cpus used by ovn kube node pods %v", ovnContainerCpus)
+				pidList, err := getOVSServicesPid(workerRTNode)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Verify ovs-vswitchd and ovsdb-server process affinity is updated
+				for _, pid := range pidList {
+					cmd := []string{"taskset", "-pc", pid}
+					cpumask, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
+					Expect(err).ToNot(HaveOccurred(), "failed to fetch cpus of %s", pid)
+					mask := strings.SplitAfter(cpumask, " ")
+					maskSet, err := cpuset.Parse(mask[len(mask)-1])
+					testlog.Infof("cpus used by ovs services are: %s", maskSet.String())
+					Expect(ovnContainerCpus).To(Equal(maskSet.String()), "affinity of ovn kube node pods(%s) do not match with ovservices(%s)", ovnContainerCpus, maskSet.String())
+					if err != nil {
+						testlog.Error(err.Error())
+					}
+				}
+				// Delete testpod1
+				testlog.Infof("Deleting pod %v", testpod1.Name)
+				deleteTestPod(testpod1)
+
+				// Check the cpus of ovnkubenode pods
+				ovnContainerCpus, err = getOvnContainerCpus(workerRTNode)
+				Expect(err).ToNot(HaveOccurred(), "Unable to fetch cpus of the containers inside ovn kubenode pods")
+				testlog.Infof("cpus used by ovn kube node pods after deleting pod %v is %v", testpod1.Name, ovnContainerCpus)
+				pidList, err = getOVSServicesPid(workerRTNode)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Verify ovs-vswitchd and ovsdb-server process affinity is updated
+				for _, pid := range pidList {
+					cmd := []string{"taskset", "-pc", pid}
+					cpumask, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
+					Expect(err).ToNot(HaveOccurred(), "failed to fetch cpus of %s", pid)
+					mask := strings.SplitAfter(cpumask, " ")
+					maskSet, err := cpuset.Parse(mask[len(mask)-1])
+					testlog.Infof("cpus used by ovs services are: %s", maskSet.String())
+					Expect(ovnContainerCpus).To(Equal(maskSet.String()), "affinity of ovn kube node pods(%s) do not match with ovservices(%s)", ovnContainerCpus, maskSet.String())
+					if err != nil {
+						testlog.Error(err.Error())
+					}
+				}
+
+				// Delete testpod2
+				deleteTestPod(testpod2)
+			})
+
+			It("[test_id:64103] ovs process affinity still excludes guaranteed pods after reboot", Label("t4"), func() {
```

Comment:
I wonder if this test is necessary. Basically we are making sure OVN-k starts and cpu manager obeys the commands even after reboot. As an e2e scenario it is probably fine, it is just expensive in terms of resources.

Reply:
Yes it's expensive , that's why we run this test case only on systems with 2 numa nodes . So mostly this test case will be executed only in downstream CI.